docs/contributing/code.md

```diff
@@ -61,6 +61,15 @@ In general, we try to avoid use of external libraries.
 - Components should **not** directly access other components -- for example, to publish to MQTT topics.
 - Components are required to dump their configuration using `ESP_LOGCONFIG` in the `dump_config()` method. This method
   is used **exclusively** to **print values** determined during `setup()` -- nothing more.
+- For time tracking, use `App.get_loop_component_start_time()` rather than `millis()`. Hardware time reads are slow
```

Comment:
:apiref: to 
https://esphome.io/api/classesphome_1_1_application (or exact App.get_loop_component_start_time member)?

Reply:
Jesse is working on moving the doxygen pages away from esphome.io and over to this domain. We can/should add this after that work is done.

Reply:
> Jesse is working on moving the doxygen pages away from esphome.io and over to this domain. 

Didn't know that. I agree, it can be updated later then.